docs/source/deploy.rst

```diff
@@ -0,0 +1,179 @@
+        if (argc != 2) {
+            std::cerr << "usage: example-app <path-to-exported-script-module>\n";
+            return -1;
+        }
+
+        // Start an interpreter manager governing 4 embedded interpreters.
+        torch::deploy::InterpreterManager manager(4);
+
+        try {
+            // Load the model from the torch.package.
+            auto package = manager.loadPackage(argv[1]);
+            auto model = package.loadPickle("model", "model.pkl");
+        } catch (const c10::Error& e) {
+            std::cerr << "error loading the model\n";
+            return -1;
+        }
+
+        std::cout << "ok\n";
+    }
+
+This small program introduces many of the core concepts of ``torch::deploy``.
+
+An ``InterpreterManager`` abstracts over a collection of independent Python
+interpreters, allowing you to load balance across them when running your code.
+
+Using the ``InterpreterManager::loadPackage`` method, you can load a
+``torch.package`` from disk and make it available to all interpreters.
+
+Finally, ``Package::loadPickle`` allows you to retrieve specific Python objects
+from the package, like the ResNet model we saved earlier.
+
+
+Building and running the application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Assuming the above C++ program was stored in a file called, `example-app.cpp`, a
+minimal CMakeLists.txt file would look like:
+
+::
+
+    cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+    project(deploy_tutorial)
+
+    find_package(Torch REQUIRED)
+
+    add_executable(example-app example-app.cpp)
+    target_link_libraries(example-app "${TORCH_LIBRARIES}")
+    set_property(TARGET example-app PROPERTY CXX_STANDARD 14)
+
+
+The last step is configuring and building the project. Assuming that our code
+directory is laid out like this:
+
+::
+
+    example-app/
+        CMakeLists.txt
+        example-app.cpp
+
+We can now run the following commands to build the application from within the
+``example-app/`` folder:
+
+::
+
+    mkdir build
+    cd build
+    # Point CMake at the built version of PyTorch we just installed.
+    SITE_PACKAGES="$(python -c 'from distutils.sysconfig import get_python_lib; print(get_python_lib())')"
+    cmake -DCMAKE_PREFIX_PATH="$SITE_PACKAGES/torch" ..
+    cmake --build . --config Release
+
+Now we can run our app:
+
+::
+
+        ./example-app /path/to/my_package.pt
+
+
+Executing ``forward`` in C++
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One you have your model loaded in C++, it is easy to execute it:
+
+::
+
+    // Create a vector of inputs.
+    std::vector<torch::jit::IValue> inputs;
+    inputs.push_back(torch::ones({1, 3, 224, 224}));
+
+    // Execute the model and turn its output into a tensor.
+    at::Tensor output = model(inputs).toTensor();
+    std::cout << output.slice(/*dim=*/1, /*start=*/0, /*end=*/5) << '\n';
+
+Notably, the model's forward function is executing in Python, in an embedded
+CPython interpreter. The model object itself is responsible for load balancing
```

Comment:
the model is not responsible for load balancing requests among interpreters, right?  The user needs to use their own request threads to make parallel requests into different interpreters.  I must be misunderstanding this part.

Reply:
I meant that multiple threads can call the same model's forward method, and the ReplicatedObject internally dispatches to different interpreters. I'll edit to clarify.